File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):
    answer = []
    citations.sort()
    n = len(citations)
    for h in range(n+1):
        ans1, ans2 = 0, 0
        for i in range(n):
            if ans2 > h:
                break
            if h <= citations[i]:
                ans1 += 1
            else:
                ans2 += 1
        if h <= ans1 and h >= ans2:
            answer.append(h)
    return max(answer)

File: 프로그래머스/lv2/42747. H－Index/H－Index.js
function solution(citations) {
    let answer = [];
    citations.sort((a, b) => a - b);
    let n = citations.length;
    for (let h = 0; h <= n; h++) {
        let [ans1, ans2] = [0, 0]
        for (let i = 0; i < n; i++) {
            if (ans2 > h) {
                break;
            } else if (h <= citations[i]) {
                ans1 += 1;
            } else {
                ans2 += 1;
            }
        }
        if (h <= ans1 && h >= ans2) {
            answer.push(h)
        }
    }

    return Math.max(...answer)
}